sources/main.cpp

```diff
@@ -333,46 +354,52 @@ std::vector<std::string> integrate_hyde_config(int argc, const char** argv) {
     hyde_flags.insert(hyde_flags.end(), cli_hyde_flags.begin(), cli_hyde_flags.end());
     clang_flags.insert(clang_flags.end(), cli_clang_flags.begin(), cli_clang_flags.end());
 
-    std::vector<std::string> result;
+    hyde_flags.push_back("--");
 
-    result.emplace_back(argv[0]);
+    command_line_args result;
 
-    result.insert(result.end(), hyde_flags.begin(), hyde_flags.end());
-
-    result.emplace_back("--");
-
-    if (!clang_flags.empty()) {
-        // it'd be nice if we could move these into place.
-        result.insert(result.end(), clang_flags.begin(), clang_flags.end());
-    }
+    result._hyde = std::move(hyde_flags);
+    result._clang = std::move(clang_flags);
 
     return result;
 }
 
 /**************************************************************************************************/
 
-std::vector<std::string> source_paths(int argc, const char** argv) {
-    return CommonOptionsParser(argc, argv, MyToolCategory).getSourcePathList();
+namespace {
+
+/**************************************************************************************************/
+
+CommonOptionsParser MakeOptionsParser(int argc, const char** argv) {
```

Comment:
The interface to `CommonOptionsParser` has changed since clang 9 - this updates to the newer interface.